docs/CircleCI.md

```diff
@@ -0,0 +1,72 @@
+
+## Partially supported properties 
+
+### jobs
+
+CircleCI `jobs` are a collection of steps that are run on a single worker unit: whether that is on a host directly, or on a virtualised host (such as within a Docker container). 
+
+The Buildkite Migration tool currently supports the following keys relative to CircleCI `jobs`:
+
+#### General
+
+- `environment`: The `job` level environment variables of a CircleCI pipeline. Applied in the generated [command step](https://buildkite.com/docs/pipelines/command-step) as [step level](https://buildkite.com/docs/pipelines/environment-variables#runtime-variable-interpolation) environment variables with the `env` key.
+- `parameters`: Reusable keys that are used within `step` definitions within a `job`. Default parameters that are specified in a `parameters` block are passed through into the [command step](https://buildkite.com/docs/pipelines/command-step)'s `commands` if specified.
+- `steps`: A collection of commands that are executed as part of a CircleCI `job`. Steps can be defined within an `alias`. All `steps` within a singular `job` are translated to the `commands` of a shared [command step](https://buildkite.com/docs/pipelines/command-step) within the generated Buildkite pipeline to ensure they share the same execution environment.
+- `working_directory`: The location of the executor where steps are run in. If set, a change directory (`cd`) command is created within the shared `commands` of a Buildkite [command step](https://buildkite.com/docs/pipelines/command-step) to the desired location.
+
+#### Docker (Execution Environment)
+
+- `docker`: Specifies that the `job` will run within a Docker container (by its `image` property) with the use of the [Docker Buildkite Plugin](https://github.com/buildkite-plugins/docker-buildkite-plugin). Additionally, the [Docker Login Plugin](https://github.com/buildkite-plugins/docker-login-buildkite-plugin) is appended if an `auth` property is defined, or the [ECR Buildkite Plugin](https://github.com/buildkite-plugins/ecr-buildkite-plugin) if an `aws-auth` property is defined within the `docker` property. Sets the [agent targeting](https://buildkite.com/docs/pipelines/defining-steps#targeting-specific-agents) for the generated [command step](https://buildkite.com/docs/pipelines/command-step) to `executor_type: docker`.
+
+#### Machine (Execution Environment)
+
+- `machine`: Specifies that the `job` will run on a machine execution environment. The `resource_class` parameter will be utilised to specify [agent targeting](https://buildkite.com/docs/pipelines/defining-steps#targeting-specific-agents) for the generated [command step](https://buildkite.com/docs/pipelines/command-step), in addition to setting the additional tags of ``executor_type: machine` and `executor_image: self-hosted`.
```

Comment:
the `resource_class` part is mentioned immediately afterwards, so this could be simplified to:
```suggestion
- `machine`: Specifies that the `job` will run on a machine execution environment. This translates to [agent targeting](https://buildkite.com/docs/pipelines/defining-steps#targeting-specific-agents) for the generated [command step](https://buildkite.com/docs/pipelines/command-step) through the  tags `executor_type: machine` and `executor_image: self-hosted`.
```

Reply:
Commited 👍